public/app/features/alerting/unified/api/buildInfo.ts

```diff
@@ -61,13 +62,22 @@ export async function discoverDataSourceFeatures(dsSettings: {
   // The current implementation of Loki's build info endpoint is useless
   // because it doesn't provide information about Loki's available features (e.g. Ruler API)
   // It's better to skip fetching it for Loki and go the Cortex path (manual discovery)
-  const buildInfoResponse = type === 'loki' ? undefined : await fetchPromBuildInfo(url);
 
+  let fallBackInfoResponse: PromBuildInfoSeriesResponse | undefined;
+  const isLoki = type === 'loki';
+  const buildInfoResponse = isLoki
+    ? undefined
+    : await fetchPromBuildInfo(url).catch(async () => {
```

Comment:
What is the HTTP `api/v1/status/buildinfo` response for Prometheus < 2.14?

Reply:
405

Reply:
except for prometheus 2.12 which returns a 404 and gets labeled as Cortex, I opened an issue for that in https://github.com/grafana/grafana/issues/54344